profiler/src/ProfilerEngine/Datadog.Profiler.Native/OpSysTools.cpp

```diff
@@ -430,20 +430,16 @@ bool OpSysTools::IsSafeToStartProfiler(double coresThreshold, double& cpuLimit)
     // For linux, we check that the wrapper library is loaded and the default `dl_iterate_phdr` is
     // the one provided by our library.
 
-    std::string wrapperLibraryName = "Datadog.Linux.ApiWrapper.x64.so";
+    const std::string wrapperLibraryName = "Datadog.Linux.ApiWrapper.x64.so";
+    const std::string wrapperLibraryNamePyroscope = "Pyroscope.Linux.ApiWrapper.x64.so";
     const std::string customFnName = "dl_iterate_phdr";
     auto* dlIteratePhdr = reinterpret_cast<void*>(::dl_iterate_phdr);
 
     Dl_info info;
     auto res = dladdr(dlIteratePhdr, &info);
     if (res == 0 || info.dli_fname == nullptr)
     {
-        wrapperLibraryName = "Pyroscope.Linux.ApiWrapper.x64.so";
```

Comment:
This if branch changes nothing, it is doing exactly the same thing as 2 lines above, so if above fails, this also fails.

In the linked issue's case it does not fail at all. It exits below at `(sharedObjectPath.filename() != wrapperLibraryName` check